Makefile

```diff
@@ -0,0 +1,30 @@
+BOARD ?= m3
+
+.PHONY: build-docker-image build-docker-image-tests setup-ude-rules \
+	    test integration-test
+
+build-docker-image:
+	docker build -t iot-lab-gateway .
+
+build-docker-image-test: build-docker-image
+	docker build -t iot-lab-gateway-tests tests
+
+setup-udev-rules:
+	sudo cp bin/rules.d/* /etc/udev/rules.d/.
+	sudo udevadm control --reload-rules
+
+test:
+	docker run -t --rm \
+		-v $(PWD):/shared \
+		-e LOCAL_USER_ID=`id -u $(USER)` \
+		iot-lab-gateway-tests tox
+
+integration-test:
+	echo $(BOARD) > $(PWD)/tests_utils/cfg_dir/board_type
+	docker run -t --rm \
+		-v $(PWD):/shared \
+		-v /dev/iotlab:/dev/iotlab \
```

Comment:
To avoid the --privileged, it would be best to do 

    --device /dev/ttynode    (the actual node, not the symlink)
    -v /dev/bus/usb/busnb/devicenb:[...]

the makefile could determine which /dev/tty and which /dev/bus/usb to share using `readlink`

Reply:
There's a tradeoff here: either we use the generic path and there's no need to pass the device path but you need the `--privileged`, either we use you proposition but then there's more logic to write in the Makefile. I'd be in favour of the former for simplicity.